### DISCUSSION TITLE
How do I start working on this?

### DISCUSSION THREAD

**Discussion by BlackRNG** [GensokyoClub/th06#18] [Q&A]
I know you can youse decomp.me to match the functions, but this is a different platform and compiler and there's not a function list yet.

**Comment by marysaka** [comment]
Hello,

We currently have a Ghidra Server and a [custom decomp.me](https://decompme.roblab.la/) instance so you need to get in touch with @roblabla on that point to get some access.

We also have a custom [discord server](https://discord.gg/VyGwAjrh9a) for the project where we discuss stuffs.

**Comment by roblabla** [comment]
Some update: we now use the official https://decomp.me server, with the "Touhou 6" preset.

The mapping file (`config/mapping.toml`) and the Ghidra Reverse (hosted at https://github.com/happyhavoc/th06-re) are now updated nightly, so they should always reflect an accurate view of where the reversing efforts are.

I'm currently working on having CI generate reports of the current functions. The next steps I think will be to figure out how I can place functions at the right offsets so that addresses (at least function addresses) match up properly.

A lot of other decomp projects start from an asm dump, and split it up and stuff. I'm a bit... reluctant to do this, mostly because it would make it a pain to keep it in sync with our ghidra database for function names, and also because it feels not-great from a copyright perspective.

---

So if you want to help, here's what you need to do:

1. Find the original game file somewhere, and store it in `resources/game.exe`.
2. Install [satsuki](https://github.com/happyhavoc/satsuki)
3. Find a function you want to reimplement. You can find the function names in `config/mapping.toml`
4. Run `scripts/generate_function_decompme <nameofthefunction>` to get an assembly dump
5. Paste the assembly dump in decompme's target asm, leave the context empty, and start decompiling.
